js/renovation/pager/pages-large.tsx

```diff
@@ -0,0 +1,143 @@
+        <Page
+          key={index.toString()}
+          index={index}
+          selected={selected}
+          onClick={onClick}
+        />
+      );
+    }
+    return (
+      <div key="delimiter" className={PAGER_PAGE_SEPARATOR_CLASS}>. . .</div>
+    );
+  });
+  return (<Fragment>{PagesMarkup}</Fragment>);
+};
+
+@ComponentBindings()
+export class LargePagesProps {
+  @OneWay() maxPagesCount?: number = 10;
+
+  @OneWay() pageCount?: number = 10;
+
+  @OneWay() pageIndex?: number = 0;
+
+  @OneWay() rtlEnabled?: boolean = false;
+
+  @Event() pageIndexChange?: (pageIndex: number) => void = () => { }; // commonUtils.noop
+}
+
+const PAGES_LIMITER = 4;
+type PageType = Partial<PageProps> | null;
+type PagesState = {
+  lastPageIndex: number;
+  pageIndexes: (number | null)[];
+  startPageIndex: number;
+} | null;
+
+function getPageIndexes(startIndex: number, pageCount: number): (number | null)[] {
+  const pageIndexes: (number | null)[] = [];
+  for (let i = 0; i < PAGES_LIMITER; i += 1) {
+    pageIndexes.push(startIndex + i);
+  }
+  if (pageIndexes[0] === 1) {
+    return [0, ...pageIndexes, null, pageCount - 1];
+  }
+  if (pageIndexes[pageIndexes.length - 1] === pageCount - 2) {
+    return [0, null, ...pageIndexes, pageCount - 1];
+  }
+  return [0, null, ...pageIndexes, null, pageCount - 1];
+}
+
+// tslint:disable-next-line: max-classes-per-file
+@Component({ defaultOptionRules: null, view: viewFunction })
+export default class LargePages extends JSXComponent(LargePagesProps) {
+  get pages(): PageType[] {
+    const { pageIndex, pageCount, maxPagesCount } = this.props as Required<LargePagesProps>;
+    let pageIndexes: (number | null)[] = [];
+    const createPage = (index: number): PageType => ({
+      index,
+      onClick: () => this.onPageClick(index),
+      selected: pageIndex === index,
+    } as PageType);
+    if (pageCount > 0) {
+      if (pageCount <= maxPagesCount) {
+        for (let i = 0; i < pageCount; i += 1) {
+          pageIndexes.push(i);
+        }
+      } else {
+        const startIndex = this.getStartIndex(pageIndex, pageCount);
+        const usePrevState = this.canUsePrevState(pageIndex, pageCount);
+        pageIndexes = usePrevState
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          ? this.prevState!.pageIndexes
+          : getPageIndexes(startIndex, pageCount);
+        if (!usePrevState) {
+          this.prevState = {
+            lastPageIndex: startIndex + PAGES_LIMITER - 1,
+            pageIndexes,
+            startPageIndex: startIndex,
+          };
+        }
+      }
+    }
+    const rtlPageIndexes = this.props.rtlEnabled ? [...pageIndexes].reverse() : pageIndexes;
+    return rtlPageIndexes.map((index) => (index === null ? null : createPage(index)));
+  }
+
+  @InternalState() private prevState = null as PagesState;
+
+  onPageClick(pageIndex: number): void {
+    if (this.props.pageIndexChange) {
+      this.props.pageIndexChange(pageIndex);
```

Comment:
`this.props.pageIndexChange?.(pageIndex);`

Reply:
Yes fix it next PR